src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +359,505 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
+            setDiskParams(diskParams => ({ ...diskParams, [key]: value }));
         }
-    }
+    };
+
+    const dialogErrorSet = (text, detail) => {
+        setDialogError(text);
+        setDialogErrorDetail(detail);
+    };
+
+    const _validationFailed = useMemo(() => {
+        const validateParams = () => {
+            const validationFailed = {};
+            const storagePoolType = diskParams.storagePool?.type;
+
+            if (mode !== CUSTOM_PATH && !storagePoolName)
+                validationFailed.storagePool = _("Please choose a storage pool");
+
+            if (mode === CUSTOM_PATH && customDiskVerificationFailed)
+                validationFailed.customPath = customDiskVerificationMessage;
 
-    dialogErrorSet(text, detail) {
-        this.setState({ dialogError: text, dialogErrorDetail: detail });
+            if (mode === CREATE_NEW) {
+                if (!diskParams.volumeName) {
+                    validationFailed.volumeName = _("Please enter new volume name");
+                }
+                if (poolTypesNotSupportingVolumeCreation.includes(storagePoolType)) {
+                    validationFailed.storagePool = cockpit.format(_("Pool type $0 does not support volume creation"), storagePoolType);
+                }
+                const poolCapacity = diskParams.storagePool && parseFloat(convertToUnit(diskParams.storagePool.capacity, units.B, diskParams.unit));
+                if (!isNaN(poolCapacity) && diskParams.size > poolCapacity) {
+                    validationFailed.size = cockpit.format(_("Storage volume size must not exceed the storage pool's capacity ($0 $1)"), poolCapacity.toFixed(2), diskParams.unit);
+                }
+            } else if (mode === USE_EXISTING) {
+                if (mode !== CUSTOM_PATH && !diskParams.existingVolumeName)
+                    validationFailed.existingVolumeName = _("Please choose a volume");
+            }
+
+            return validationFailed;
+        };
+        return validateParams();
+    }, [diskParams, mode, customDiskVerificationFailed, customDiskVerificationMessage, storagePoolName]);
+    const validationFailed = validate ? _validationFailed : {};
+
+    let defaultBody;
+    if (dialogLoading) {
+        defaultBody = (
+            <Bullseye>
+                <Spinner />
+            </Bullseye>
+        );
+    } else if (isMediaInsertion) {
+        defaultBody = (
+            <Form onSubmit={e => e.preventDefault()} isHorizontal>
+                <FormGroup fieldId={`${idPrefix}-source`}
+                           id={`${idPrefix}-source-group`}
+                           label={_("Source")} isInline hasNoPaddingTop>
+                    <Radio id={`${idPrefix}-custompath`}
+                           name="source"
+                           label={_("Custom path")}
+                           isChecked={mode === CUSTOM_PATH}
+                           onChange={() => onValueChanged('mode', CUSTOM_PATH)} />
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231116-154931-f2ab7bbc-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)